Parser breaks on `ORDER BY Timestamp`

A query like this `SELECT Timestamp FROM events ORDER BY Timestamp` makes the parser panic:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x101583b]

goroutine 19 [running]:
testing.tRunner.func1.2({0x1392880, 0x21fe600})
	external/go_sdk/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	external/go_sdk/src/testing/testing.go:1634 +0x377
panic({0x1392880?, 0x21fe600?})
	external/go_sdk/src/runtime/panic.go:770 +0x132
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseColumnExpr(0xc0000a2f10, 0x26)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:337 +0x7b
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseUnaryExpr(0xc0000a2f10, 0x26)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:304 +0x2c5
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseMulDivModExpr(0xc000151f10, 0xc00039bf80?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:229 +0x1c
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseAddSubExpr(0xc000151f10, 0x7e4e636825b8?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:182 +0x1c
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseCompareExpr(0xc000151f10, 0x26)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:123 +0x27
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseIsOrNotNull(0xc000151f10, 0x26)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:96 +0x3b
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseNotExpr(0xc0000a2f10, 0x1d)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:82 +0xe7
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseAndExpr(0xc000151f10, 0xc0000a2b40?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:59 +0x1c
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseOrExpr(0xc000151f10, 0x3fc4f0?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:37 +0x1c
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseExpr(0xc000151f10, 0xc0000a2c28?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_column.go:16 +0x1c
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseOrderExpr(0xc000151f10, 0x1d)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_table.go:641 +0x25
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseOrderByClause(0xc000151f10, 0x1d)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_table.go:619 +0x7b
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).tryParseOrderByClause(0xc0000a2f10, 0x1d)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_table.go:612 +0x67
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseSelectStmt(0xc000151f10, 0x0)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_query.go:871 +0x479
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseSelectQuery(0xc000151f10, 0x1583bd5?)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_query.go:754 +0x159
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).parseStmt(0xc000151f10, 0x0)
	/home/joakim/go/pkg/mod/github.com/!after!ship/clickhouse-sql-parser@v0.3.4-0.20240627110054-61b3867d9915/parser/parser_table.go:939 +0x14d
github.com/AfterShip/clickhouse-sql-parser/parser.(*Parser).ParseStmts(0xc000151f10)
```

--- a/parser/parser_column.go
+++ b/parser/parser_column.go
@@ -298,7 +298,7 @@ func (p *Parser) parseColumnExpr(pos Pos) (Expr, error) { //nolint:funlen
 		if err != nil {
 			return nil, err
 		}
-		if nextToken.Kind == TokenString {
+		if nextToken != nil && nextToken.Kind == TokenString {
 			return p.parseString(pos)
 		}
 		return p.parseIdentOrFunction(pos)

--- a/parser/testdata/query/format/select_order_by_timestamp.sql
+++ b/parser/testdata/query/format/select_order_by_timestamp.sql
@@ -1,0 +1,10 @@
+-- Origin SQL:
+SELECT Timestamp FROM events ORDER BY Timestamp;
+
+-- Format SQL:
+
+SELECT 
+  Timestamp
+FROM
+  events
+ORDER BY Timestamp;

--- a/parser/testdata/query/output/select_order_by_timestamp.sql.golden.json
+++ b/parser/testdata/query/output/select_order_by_timestamp.sql.golden.json
@@ -1,0 +1,74 @@
+[
+  {
+    "SelectPos": 0,
+    "StatementEnd": 47,
+    "With": null,
+    "Top": null,
+    "SelectColumns": {
+      "ListPos": 7,
+      "ListEnd": 16,
+      "HasDistinct": false,
+      "Items": [
+        {
+          "Name": "Timestamp",
+          "QuoteType": 1,
+          "NamePos": 7,
+          "NameEnd": 16
+        }
+      ]
+    },
+    "From": {
+      "FromPos": 17,
+      "Expr": {
+        "Table": {
+          "TablePos": 22,
+          "TableEnd": 28,
+          "Alias": null,
+          "Expr": {
+            "Database": null,
+            "Table": {
+              "Name": "events",
+              "QuoteType": 1,
+              "NamePos": 22,
+              "NameEnd": 28
+            }
+          },
+          "HasFinal": false
+        },
+        "StatementEnd": 28,
+        "SampleRatio": null,
+        "HasFinal": false
+      }
+    },
+    "ArrayJoin": null,
+    "Window": null,
+    "Prewhere": null,
+    "Where": null,
+    "GroupBy": null,
+    "WithTotal": false,
+    "Having": null,
+    "OrderBy": {
+      "OrderPos": 29,
+      "ListEnd": 47,
+      "Items": [
+        {
+          "OrderPos": 29,
+          "Expr": {
+            "Name": "Timestamp",
+            "QuoteType": 1,
+            "NamePos": 38,
+            "NameEnd": 47
+          },
+          "Direction": "None"
+        }
+      ]
+    },
+    "LimitBy": null,
+    "Limit": null,
+    "Settings": null,
+    "Format": null,
+    "UnionAll": null,
+    "UnionDistinct": null,
+    "Except": null
+  }
+]

--- a/parser/testdata/query/select_order_by_timestamp.sql
+++ b/parser/testdata/query/select_order_by_timestamp.sql
@@ -1,0 +1,1 @@
+SELECT Timestamp FROM events ORDER BY Timestamp;

